depthai_ros_driver/include/depthai_ros_driver/conversion.hpp

```diff
@@ -99,16 +114,26 @@ struct adapt_dai2ros<depthai_datatype_msgs::RawImgFrame> {
         bridge.header.stamp.sec = input.ts.sec;
         bridge.header.stamp.nsec = input.ts.nsec;
 
-        pub.image_pub.publish(bridge.toImageMsg());
-        sensor_msgs::CameraInfo info;  // @TODO(kunaltyagi): get camera info from somewhere
-        pub.camera_info_pub.publish(info);
+        // @TODO(kunaltyagi): switch between raw and compressed publishers
```

Comment:
@ardiya How can I get this information?

Reply:
`template <class MsgType, bool is_t_compressed=false>` in adapt_dai2ros and in generate_pub_lambda?

Then call generate_pub_lambda when it's node is coming from dai::node::VideoEncoder? 
```
                case dai::DatatypeEnum::ImgFrame:
                if (node_is_coming_from_encoder) {
                  generate_pub_lambda<depthai_datatype_msgs::RawImgDetections, true>(...)
                } else {
                  generate_pub_lambda<depthai_datatype_msgs::RawImgDetections, false>(...)
                }
```
It feels like you told me this last time, no?

Reply:
Can this be inferred at runtime as well?

Reply:
if you use std::any, it's possible to have both image and compressedimage at runtime? I never really use it though

Reply:
I was wondering if the output we read can be used to identify if it's compressed

Reply:
Oh, I haven't done that but the idea is compressed image is `dai::RawImgFrame::Type::BITSTREAM` type.
https://github.com/rapyuta-robotics/depthai_ros/blob/gen2_ros/depthai_ros_driver/src/conversion.cpp#L94-L98
I haven't handle this as special case yet.